priorityQueues/priorityQueue.py

```diff
@@ -0,0 +1,65 @@
+"""
+This class is an implementation of a (minimum) priority queue using an array backing. 
+It holds on to data as a list of tuples: (data, priority)
+
+"""
+from collections import deque
+
+class PriorityQueue():
+
+    def __init__(self):
+        self.queue = []
+
+    """Push item onto heap, maintaining the heap invariant afterwards."""
+    def push(self, item):
+        if type(item) != tuple or type(item[1]) != int:
+            raise ValueError('Item to insert must be a tuple of (data, int(priority))')
+        self.queue.append(item)
+        self.fixup_queue(0, len(self.queue)-1)
+
+    """
+    Borrowed from the python heap implementation here https://hg.python.org/cpython/file/3.3/Lib/heapq.py
```

Comment:
You should implement the fix-up yourself. 
If you need to write your own heap, you will need to know how to move from the root @ index 0 to a leaf and vice- versa.
It involves simply multiplying and dividing by 2 and some addition and subtraction depending

[0,1,2,3,4,5,6]
               0
              /  \
            1      2
          /   \   /  \
        3    4  5   6